k4FWCore/src/PodioDataSvc.cpp

```diff
@@ -109,14 +123,19 @@ StatusCode PodioDataSvc::i_setRoot(std::string root_path, DataObject* pRootObj)
 }
 
 void PodioDataSvc::endOfRead() {
+  m_eventNum++;
+
+  if (!m_bounds_check_needed) {
+    return;
+  }
+
   StatusCode sc;
-  if (m_eventMax != -1) {
-    if (m_eventNum++ >= m_eventMax - 1) {  // we start counting at 0 thus the -1.
-      info() << "Reached end of file with event " << m_eventMax << endmsg;
-      IEventProcessor* eventProcessor;
-      sc = service("ApplicationMgr", eventProcessor);
-      sc = eventProcessor->stopRun();
-    }
+  // m_eventNum already points to the next event here so check if it is available
+  if (m_eventNum >= m_numAvailableEvents) {
+    info() << "Reached end of file with event " << m_eventNum << "/" << m_numAvailableEvents << endmsg;
```

Comment:
I hate to keep adding on this but this comment won't run when your request to read exactly the number of events that a file has, only with `EvtMax = -1` or when you request to read more events than available. Therefore it's a bit of redundant information because when it will appear it will always print "Reached ... N/N" where N is the number of events

Reply:
do you have a change suggestion?

Reply:
Not really, I only think it would be useful when you request more events than available and it tells you that it has stopped. Other than that sometimes it may appear (with `EvtMax = -1`) or not (when reading N events with N available events) when reaching the end of the file so it's a bit inconsistent. It would be possible by moving the info message above the return in `endOfRead()` so that it also runs in that last case when reading N with N available

Reply:
I changed it to a more meaningful message:
"Reached end of file with event N (M events requested)"